profiler/tools/stencil_reproducer.py

```diff
@@ -0,0 +1,135 @@
+import fnmatch
+import pickle
+from datetime import datetime
+from glob import glob
+from os import getcwd, getenv, listdir, mkdir, path, walk
+from shutil import copy, copytree
+from typing import Any, Dict, Tuple
+
+import numpy as np
+from gt4py import storage
+
+
+STENCIL_CANDIDATE_FOR_EXTRACT: Dict[str, Tuple[str, str]] = {}
+STENCIL_SERIALIZED_ARGUMENTS: Dict[str, Any] = {}  # Indexed on file path
+
+
+def field_serialization(frame, event, args):
+    """Serialize all fields from a stencil"""
+    if event == "call" or event == "return":
+        for stencil_key, stencil_info in STENCIL_CANDIDATE_FOR_EXTRACT.items():
+            if (
+                frame.f_code.co_name == "run"
+                and stencil_info[0] == frame.f_code.co_filename
+            ):
+                print(f"[PROFILER] Pickling args of {stencil_key} @ event {event}")
+                if event == "call":
+                    prefix = "pre_run_"
+                else:
+                    prefix = "post_run_"
+
+                scalars = {}
+                for arg_key, arg_value in frame.f_locals.items():
+                    if arg_key == "self":
+                        continue
+                    if isinstance(arg_value, storage.Storage):
+                        arg_value.device_to_host()
+                        np.savez_compressed(
+                            f"{stencil_info[1]}/data/{prefix}_{arg_key}.npz",
+                            arg_value.data,
+                        )
+                    else:
+                        scalars[arg_key] = arg_value
+                scalar_file = f"{stencil_info[1]}/data/{prefix}_scalars.pickled"
+                with open(scalar_file, "wb") as handle:
+                    pickle.dump(scalars, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+
+def collect_stencil_candidate(stencil_name):
+    """Collect all stencils that correspond to `stencil_name`.
+
+    Multiple stencils can be collected if compile time varaiables lead to multiple
+    stencil code path.
+    """
+    expected_py_wrapper_partialname = f"m_{stencil_name}__"
+    gt_cache_root = getenv("GT_CACHE_ROOT")
+    gt_cache_root = gt_cache_root if gt_cache_root is not None else getcwd()
+    print(f"[PROFILER] Searching for {stencil_name} in {gt_cache_root}...")
+    for fname in listdir(gt_cache_root):
+        fullpath = path.join(gt_cache_root, fname)
+        if fname.startswith(".gt_cache") and path.isdir(fullpath):
+            for root, _dirnames, filenames in walk(fullpath):
```

Comment:
```suggestion
            for root, _, filenames in walk(fullpath):
```

Reply:
So what is the proper way of doing this? The `_var` is legit, is that a team guideline to use `_`?

(note that I have very little care for it :D)

Reply:
more of a python standard

https://luacheck.readthedocs.io/en/0.12.0/warnings.html#unused-variables-and-values

Reply:
Will do!